client/components/LearnerViewDirective.js

```diff
@@ -21,97 +21,116 @@ tie.directive('learnerView', [function() {
+                    <option style="display: none" value="">Theme</option>
+                    <option></option>
+                  </select>     
+                  <button class="tie-code-reset" name="code-reset"
+                      ng-click="resetCode()">
+                    Reset Code
+                  </button>
+                  <div class="tie-code-auto-save" ng-show="autosaveTextIsDisplayed">
+                    Saving code now...
+                  </div>
+                  <button class="tie-run-button"
+                      ng-class="{'active': !nextButtonIsShown}"
+                      ng-click="submitCode(code)"
+                      ng-disabled="nextButtonIsShown">
+                    Run
+                  </button>
+                  <div class="tie-next-curtain-container"
+                      ng-if="nextButtonIsShown">
+                    <div class="tie-next-curtain"></div>
+                    <div class="tie-arrow-highlighter"></div>
+                    <div ng-click="showNextTask()" class="tie-next-arrow">
+                      <span class="tie-next-button-text">Next</span>
+                    </div>
                   </div>
                 </div>
               </div>
             </div>
-          </div>
-          <div class="tie-question-ui">
-            <div class="tie-question-window">
-              <h3 class="tie-question-title">{{title}}</h3>
-              <div class="tie-previous-instructions">
-                <div ng-repeat="previousInstruction in previousInstructions track by $index">
-                  <p ng-repeat="paragraph in previousInstruction track by $index">{{paragraph}}</p>
-                  <hr>
+            <div class="tie-question-ui">
+              <div class="tie-question-window" ng-class="{'night-mode': isInDarkMode}">
+                <h3 class="tie-question-title">{{title}}</h3>
+                <div class="tie-previous-instructions">
+                  <div ng-repeat="previousInstruction in previousInstructions track by $index">
+                    <p ng-repeat="paragraph in previousInstruction track by $index">{{paragraph}}</p>
+                    <hr>
+                  </div>
+                </div>
+                <div class="tie-instructions">
+                  <p ng-repeat="paragraph in instructions">{{paragraph}}</p>
                 </div>
-              </div>
-              <div class="tie-instructions">
-                <p ng-repeat="paragraph in instructions">{{paragraph}}</p>
               </div>
             </div>
           </div>
         </div>
       </div>
       <style>
+        html {
+          height: 100%;
+        }
         body {
           background-color: rgb(242, 242, 242);
           font-family: Roboto, 'Helvetica Neue', 'Lucida Grande', sans-serif;
           font-size: 15px;
+          height: 100%;
+          margin: 0px;
+        }
+        .tie-wrapper {
+          height: 100%;
+        }
```

Comment:
For the css style statements, please order alphabetically, unless it impacts cascading.